tests/net_dynamic_hb.rs

```diff
@@ -269,10 +277,25 @@ fn do_drop_and_readd(cfg: TestConfig) {
                 }
             }
         }
-        if removed_ourselves {
-            // Further operations on the cranked node are not possible. Continue with processing
-            // other nodes.
-            continue;
+
+        let (era, hb_epoch) = state.get_net_mut()[node_id].algorithm().algo().epoch();
+        if node_id != pivot_node_id
+            && awaiting_addition_input.contains(&node_id)
+            && state.shutdown_epoch.is_some()
+            && era + hb_epoch == state.shutdown_epoch.unwrap()
```

Comment:
That restriction shouldn't be necessary, I think? We can vote for re-adding in any epoch.

Reply:
It fails without the validators waiting for this epoch. Is this incorrect, you think?

Reply:
Right, I'd just use `>=` then. :+1: